spec/tinymce_examples_spec.rb

```diff
@@ -0,0 +1,313 @@
+require_relative 'spec_helper'
+
+describe HTMLToConfluenceParser, "when consuming TinyMCE generated HTML" do
+
+  context "edge case with nested formats with spans and &nbsp;" do
+
+    it "should handle nested formatting" do
+      html = <<~HTML
+        test<i>test<b>test</b></i>
+      HTML
+
+      markup = <<~MARKUP
+        test_test*test*_
+      MARKUP
+
+      expect(html).to match_markup(markup)
+    end
+
+    it "should handle arbitrary spans" do
+      html = <<~HTML
+        test<span>test<span>test</span></span>
+      HTML
+
+      markup = <<~MARKUP
+        testtesttest
+      MARKUP
+
+      expect(html).to match_markup(markup)
+    end
+
+    it "should handle arbitrary %nbsp;" do
+      html = <<~HTML
+        test&nbsp;test&nbsp;test
+      HTML
+
+      markup = <<~MARKUP
+        test test test
+      MARKUP
+
+      expect(html).to match_markup(markup)
+    end
+
+    it "should handle trailing %nbsp; inside formats" do
+      html = <<~HTML
+        <b>test&nbsp;</b>
+      HTML
+
+      markup = <<~MARKUP
+        *test *
```

Comment:
The only way I can get this formulation (generated by tinymce when applying formatting to highlighted text or nested formats across words, IIRC) to work in JIRA is something like: `*test&nbsp;{*}test`. TinyMCE does give us the nbsp in these contexts but we throw it away for a normal space.

Alternatives like shifting the padding spaces outside the tags would force an extra auto-update loop between JIRA and Aha! to normalize.